File: app/src/main/java/com/andrespaez/bitrise/ui/adapters/list/models/GenericCategoryItemAbstract.java
package com.andrespaez.bitrise.ui.adapters.list.models;

public class GenericCategoryItemAbstract implements GenericCategoryItem<Object> {

    private Object data;
    private int type;
    private String category;

    public GenericCategoryItemAbstract(Object data, String category) {
        this.data = data;
        this.category = category;
    }

    public GenericCategoryItemAbstract(Object data, int type, String category) {
        this.data = data;
        this.type = type;
        this.category = category;
    }

    public Object getData() {
        return this.data;
    }

    public int getType() {
        return this.type;
    }

    public String getCategoryName() {
        return this.category;
    }

    public int compareTo(GenericCategoryItem categoryItem) {
        return this.category.compareTo(categoryItem.getCategoryName());
    }
}

File: app/src/main/java/com/bitrise/app/ui/adapters/list/models/GenericCategoryItemAbstract.kt
package com.bitrise.app.ui.adapters.list.models

class GenericCategoryItemAbstract(
    override var data: Any,
    override val categoryName: String,
    override val type: Int = 0,
    override val weight: Int = 0,
) : GenericCategoryItem<Any> {

    override fun compareTo(var1: GenericCategoryItem<*>): Int {
        return categoryName.compareTo(var1.categoryName)
    }
}